src/sage/misc/latex.py

```diff
@@ -485,16 +485,35 @@ def default_engine():
 
         sage: from sage.misc.latex import default_engine
         sage: default_engine()  # random
-        ('lualatex', 'LuaLaTeX')
+        'lualatex'
+
+    TESTS::
+
+    Ensure that this (expensive) function is not necessary to obtain
+    the latex representation of a matrix (doing so probes the latex
+    options dict for the delimiters)::
+
+        sage: import sage.misc.latex
+        sage: real_de = sage.misc.latex.default_engine
+        sage: def crash():
+        ....:     raise ValueError
+        sage: sage.misc.latex.default_engine = crash
+        sage: latex(matrix.identity(QQ, 2))
+        \left(\begin{array}{rr}
+        1 & 0 \\
+        0 & 1
+        \end{array}\right)
+        sage: sage.misc.latex.default_engine = real_de
+
     """
     from sage.features.latex import pdflatex, xelatex, lualatex
     if lualatex().is_present():
-        return 'lualatex', 'LuaLaTeX'
+        return 'lualatex'
```

Comment:
Looks like dangerous API change to me. Is this function public API? (it doesn't even start with an underscore?)

Reply:
It's one year old, not in the default namespace and not used outside of `sage.misc.latex`. Obviously you're right, but I bet I could give it always wanted and no one would notice.

Reply:
I added the underscore

Reply:
@kwankyu are you OK with me making `default_engine()` an internal/underscore method in `sage.misc.latex`?

Reply:
No.  I think converting a public function to an internal function is an even bigger API change. Though "no one would notice", I see no reason to treat it as an exception to the deprecation principle.

On the other hand, I don't remember why the latex option list contains "engine_name", which is used nowhere within Sage. I agree that it is better to drop it as it is not really an option.

I suggest to leave `default_engine()` in public API, but change its behavior dropping the engine name. If you do so, please follow the proper deprecation procedure. If it is cumbersome, how about leaving `default_engine()` as it is, but just dropping "engine_name" from the latex option list?

Reply:
I guess I will put the old `default_engine` (no underscore) back, and deprecate it. Then we will have `_default_engine` witho no "engine_name", internal from now on, but without breaking the API for anyone.

Reply:
OK.